crates/matrix-sdk/src/client/builder.rs

```diff
@@ -99,7 +99,7 @@ impl ClientBuilder {
     /// This method is mutually exclusive with
     /// [`homeserver_url()`][Self::homeserver_url], if you set both whatever was
     /// set last will be used.
-    pub fn user_id(self, user_id: &UserId) -> Self {
+    pub fn user_id_server(self, user_id: &UserId) -> Self {
```

Comment:
I think the name `user_id` is misleading because it sets the name of the server of this user id, but looks as if it was configuring the user id.

Reply:
I agree that it's a bit misleading, but `user_id_server` doesn't seem ideal either. Maybe `homeserver_from_user_id`?

Reply:
Note sure I understand this suggestion, isn't that the same method as [this existing](https://github.com/matrix-org/matrix-rust-sdk/blob/main/crates/matrix-sdk/src/client/builder.rs#L111) one? I am also open to other name suggestions, but as mentioned before `user_id` really does not communicate the purpose: as you set `clientBuilder.user_id(` it appears that you are really setting the user id instead of server.